presto-spi/src/main/java/com/facebook/presto/spi/constraints/TableConstraint.java

```diff
@@ -36,20 +34,22 @@
 public abstract class TableConstraint<T>
```

Comment:
Does this class even need to be parameterized over `T`? I couldn't find anywhere else where we use something other than the column name as a `String`. This might not be a bad time to simplify?